### PR TITLE
Fix benchmark not to assume the input is a list of tensor.

### PR DESCRIPTION
**Pull request by vanbasten23**
Currently, it fails with error:
```
Traceback (most recent call last):
  File "xla/benchmarks/experiment_runner.py", line 968, in <module>
    main()
  File "xla/benchmarks/experiment_runner.py", line 964, in main
    runner.run()
  File "xla/benchmarks/experiment_runner.py", line 61, in run
    self.run_single_config()
  File "xla/benchmarks/experiment_runner.py", line 249, in run_single_config
    benchmark_model = self.model_loader.load_model(model_config,
  File "/ansible/pytorch/xla/benchmarks/benchmark_model.py", line 59, in load_model
    benchmark_model.prepare_for_experiment(
  File "/ansible/pytorch/xla/benchmarks/benchmark_model.py", line 160, in prepare_for_experiment
    assert self.example_inputs[0].device.type.lower(
AttributeError: 'list' object has no attribute 'device'
```

It turns out the `self.example_inputs` may not be a list of tensors. 

Test: 
`# CUDA_VISIBLE_DEVICES=0 ZERO_COPY_ENABLED=1 python xla/benchmarks/experiment_runner.py     --suite-name=torchbench     --accelerator=cuda     --progress-bar      --model-config=\{\"model_name\":\"detectron2_fasterrcnn_r_101_c4\"\}     --experiment-config=\{\"accelerator\":\"cuda\",\"xla\":\"PJRT\",\"xla_flags\":null,\"dynamo\":\"openxla\",\"torch_xla2\":null,\"test\":\"train\"\,\"keep_model_data_on_cuda\":true\}     --repeat 1`

### CODE REVIEW

**Comment by zpcore** [comment]
Probably I add `self.example_inputs[0]` before. This is to handle the update in the upstream torchbench, where the example_input of hugging face models were updated to a list. Did you try with the latest pytorch/benchmark?

**Comment by zpcore** [comment]
IIRC, example_inputs will be a list only for hugging face models.

**Comment by vanbasten23** [comment]
> Probably I add `self.example_inputs[0]` before. This is to handle the update in the upstream torchbench, where the example_input of hugging face models were updated to a list. Did you try with the latest pytorch/benchmark?

Actually, I added it in https://github.com/pytorch/xla/pull/7193/files#diff-0b5628ca13929a70b4e70f83edc88a27558320c0b84f9967df5ce63498d3ef13R159-R162 lol. But I've update it to use pytree

**Review by zpcore — APPROVED** [review]
LGTM!

File: benchmarks/benchmark_model.py [hunk 1]
```diff
@@ -157,8 +157,12 @@ def prepare_for_experiment(self, dynamo_compilation_opts):
...
 
     if keep_model_data_on_cuda:
-      assert self.example_inputs[0].device.type.lower(
-      ) == 'cuda', 'When keep_model_data_on_cuda is set, the input data should remain on the CUDA device.'
+
+      def assert_func(t):
+        assert t.device.type.lower(
+        ) == 'cuda', 'When keep_model_data_on_cuda is set, the input data should remain on the CUDA device.'
+
+      pytree.tree_map_only(torch.Tensor, assert_func, self.example_inputs)
```
Comment (zpcore) [inline benchmarks/benchmark_model.py:165]: I am not sure if torch.Tensor type apply to jax tensor or not. Maybe this doesn't work for torch_xla2 if they want to keep model data on cuda also.

**Comment by vanbasten23** [comment]
Thanks @zpcore for the review!

**Pull request merged**